### PR TITLE
Update/structure improvements

### PR DESCRIPTION
**Pull request by collectedmind**
## Goal

Revisiting the extension structure armed with knowledge from additional reactor work.  This brings the extension in line with the rest of the front end refactor work.

## Todos:
- [ ] - Remove external build dependency in favor of a local config
- [ ] - Add Storybooks for all UI elements 
- [ ] - Set up test structure
- [ ] - Update Structure for Components/Containers/Actions/Reducers/Sagas
- [ ] - Update how state is connected to components
- [ ] - Improve code structure consistency
- [ ] - Remove Sass dependency in favor of CSS-in-JS with emotion.sh
- [ ] - Update panel loading to be more agnostic
- [ ] - Improve server sync

## Implementation Decisions
This was an exercise in revisiting and simplifying code structures so they are easier to work with, easier to reason about, and more predictable.  This also is more in line with upcoming refactor work on the web app.

## All Submissions:

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### CODE REVIEW

File: src/Components/Modules/Tagging/tagging.stories.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { Tagging } from './tagging'
+import { COLORS } from 'Common/_mocks/colors'
+
+/* Tagging Story
+------------------------------------------------------- */
+export default class MockState extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      tags: samplesize(COLORS, 5),
```
Comment (nelsonomuto) [inline src/Components/Modules/Tagging/tagging.stories.js]: Hey @collectedmind I am setting up storyshots and it runs all the stories and creates snapshots. Each time we run though there is a new random sample of colors.

May we use COLORS.slice(0, 5) instead to keep our tests deterministic

Comment (collectedmind) [reply src/Components/Modules/Tagging/tagging.stories.js]: Certainly.  Feel free to update that as need be.  The main reason it is randomized was so we get a sampling of different tag lengths.

File: src/Components/Views/Recommendations/List/list.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  padding: 0;
+`
+
+export class RecommendationList extends React.Component {
+  render() {
+    const { items } = this.props
+    return items.length ? (
+      <ListWrapper>
+        {items.map(item => (
+          <RecommendationItem key={item.item_id} item={item} />
```
Comment (nelsonomuto) [inline src/Components/Views/Recommendations/List/list.js]: should this key be `item.fee_item_id` and not `item.item_id` @collectedmind 

screen shot 2018-07-16 at 1 21 30 pm

screen shot 2018-07-16 at 1 21 23 pm

File: src/Components/Modules/Tagging/tagging.stories.js (hunk 1 above)
Comment (nelsonomuto) [reply src/Components/Modules/Tagging/tagging.stories.js]: I am planning on switching this to use slice and deprecate the lodash.samplesize in a subsequent PR @collectedmind

File: src/Components/Views/Survey/survey.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      </FieldOption>
+    )
+  })
+}
+
+function Fields({ fields }) {
+  return fields.map(fieldset => {
+    const { label, options, name } = fieldset
+    return (
+      <FieldSet>
```
Comment (nelsonomuto) [inline src/Components/Views/Survey/survey.js]: keys required for FieldSet items

screen shot 2018-07-16 at 1 27 27 pm
screen shot 2018-07-16 at 1 30 48 pm
screen shot 2018-07-16 at 1 30 40 pm

File: src/Components/Views/Recommendations/List/list.js (hunk 2 above)
Comment (collectedmind) [reply src/Components/Views/Recommendations/List/list.js]: 👍

File: src/Components/Views/Survey/survey.js (hunk 3 above)
Comment (collectedmind) [reply src/Components/Views/Survey/survey.js]: Thanks!  I will update those two key issues.

File: src/Components/Views/Survey/survey.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  width: 100%;
+  margin-bottom: 4px;
+  font-size: 0.875rem;
+`
+
+function FieldOptions({ options, name }) {
+  return options.map(option => {
+    const { text, value } = option
+    return (
+      <FieldOption onClick={() => console.log(`${name}:${value}`)}>
```
Comment (nelsonomuto) [inline src/Components/Views/Survey/survey.js]: a unique key is required, there is no unique property in the dataset so using index* (don't forsee any option being dynamically updated)

screen shot 2018-07-17 at 10 10 52 am

File: src/Components/Elements/Colors/colors.stories.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+storiesOf('Elements|Colors', module)
+  .addDecorator(story => <div>{story()}</div>)
+  .add('Colors', () => displayColors({ Colors, Shades }))
+
+function displayColors(colorObject) {
+  return Object.keys(colorObject).map(colorType => {
+    const colors = colorObject[colorType]
+
+    return (
+      <div>
```
Comment (nelsonomuto) [inline src/Components/Elements/Colors/colors.stories.js]: Needs a key 

screen shot 2018-07-19 at 8 18 43 am

File: src/Components/Elements/Icons/icon.stories.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+          background-color: ${Colors.teal};
+          z-index: -100;
+        }
+      `}>
+      {story()}
+    </div>
+  ))
+  .add('All', () => (
+    <div style={{ fontSize: '128px', lineHeight: '1.25em' }}>
+      {extensionIcons.map(icon => <Icon name={icon} key={icon.name} />)}
```
Comment (nelsonomuto) [inline src/Components/Elements/Icons/icon.stories.js]: Needs unique key
screen shot 2018-07-19 at 8 22 35 am
screen shot 2018-07-19 at 8 22 43 am

**Comment by collectedmind** [comment]
Closing this as it is a bit dated now. We will be revisiting this work as we reconcile the safari build with the chrome work.

**Pull request closed without merging**